### PR TITLE
Permissions and Custom Claims controller

### PR DESCRIPTION
**Pull request by landonpierce**
Added routes to support b2c custom claims enrichment and basic CRUD operations for permissions. Still need to add some polish and validation around some of the routes, as well as some swagger attributes to support documentation.

Also added github actions for deployment to permissions api

### CODE REVIEW

File: src/Saas.Permissions/Saas.Permissions.Api/Controllers/PermissionsController.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        [Route("RemoveUserPermissionsFromTenant")]
+        public async Task<IActionResult> RemoveUserPermissionsFromTenant(Guid tenantId, Guid userId, string[] permissions)
+        {
+            await _permissionsService.RemoveUserPermissionsFromTenantAsync(tenantId, userId, permissions);
+            return Ok();
+        }
+
+        [HttpGet]
+        [Route("GetTenantsForUser")]
+        public async Task<ICollection<Guid>> GetTenantsForUser(Guid userId, string? filter)
```
Comment (asanjabi) [inline src/Saas.Permissions/Saas.Permissions.Api/Controllers/PermissionsController.cs:50]: For filter I was thinking specific roles, but we should wait to see if we actually need it.

**Review by asanjabi — CHANGES_REQUESTED** [review]
- Change TenantId to string
- Fix the warnings

**Comment by landonpierce** [comment]
@asanjabi all changes are in. I think you have to re-review and approve before I can merge. No rush, thanks.

**Review by asanjabi — APPROVED** [review]

**Comment by asanjabi** [comment]
@landonpierce done

**Pull request merged**